modules/Bio/EnsEMBL/DBSQL/SequenceAdaptor.pm

```diff
@@ -80,64 +69,14 @@ our $SEQ_CACHE_MAX   = (2 ** $SEQ_CHUNK_PWR) * $SEQ_CACHE_SZ;
 =cut
 
 sub new {
-  my $caller = shift;
-
+  my ($caller, $db, $chunk_power, $cache_size) = shift;
```

Comment:
Yes, you initialise $caller and declare the rest in one shot, but I don't see where the rest of the array variables are initialised.